GitFyle.Core.Api.Tests.Unit/Services/Foundations/Contributions/ContributionServiceTests.Exceptions.RemoveById.cs

```diff
@@ -0,0 +1,169 @@
+            SqlException sqlException = CreateSqlException();
+
+            var failedContributionStorageException =
+                new FailedStorageContributionException(
+                    message: "Failed contribution storage error occurred, contact support.",
+                        innerException: sqlException);
+
+            var expectedContributionDependencyException =
+                new ContributionDependencyException(
+                    message: "Contribution dependency error occurred, contact support.",
+                        innerException: failedContributionStorageException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectContributionByIdAsync(someContributionGuid))
+                    .ThrowsAsync(sqlException);
+
+            // when
+            ValueTask<Contribution> removeContributionByIdTask =
+                this.contributionService.RemoveContributionByIdAsync(someContributionGuid);
+
+            ContributionDependencyException actualContributionDependencyException =
+                await Assert.ThrowsAsync<ContributionDependencyException>(
+                    removeContributionByIdTask.AsTask);
+
+            // then
+            actualContributionDependencyException.Should().BeEquivalentTo(
+                expectedContributionDependencyException);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectContributionByIdAsync(It.IsAny<Guid>()),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogCriticalAsync(It.Is(SameExceptionAs(
+                    expectedContributionDependencyException))),
+                        Times.Once);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffsetAsync(),
+                    Times.Never);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        private async Task ShouldThrowDependencyValidationExceptionOnRemoveByIdIfDbConcurrencyOccursAndLogItAsync()
+        {
+            // given
+            Guid someContributionId = Guid.NewGuid();
+
+            var dbUpdateConcurrencyException =
+                new DbUpdateConcurrencyException();
+
+            var lockedContributionException =
+                new LockedContributionException(
+                    message: "Locked contribution record error occurred, please try again.",
+                    innerException: dbUpdateConcurrencyException);
+
+            var expectedContributionDependencyValidationException =
+                new ContributionDependencyValidationException(
+                    message: "Contribution dependency validation error occurred, fix errors and try again.",
+                    innerException: lockedContributionException);
```

Comment:
For dependency validations, you sould always assign the data dictionary to the localize exception so that services higher up the chain can have access to it.
The unwrap/wrap logic only strips of the catagorization exception.

```suggestion
            var expectedContributionDependencyValidationException =
                new ContributionDependencyValidationException(
                    message: "Contribution dependency validation error occurred, fix errors and try again.",
                    innerException: lockedContributionException,
                    data: lockedContributionException.Data);
```